src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -108,6 +107,35 @@ impl ConvertShares {
     }
 }
 
+pub async fn convert_all_shares<F: Field>(
+    ctx: ProtocolContext<'_, F>,
+    input: &[u64],
+) -> Result<Vec<Vec<Replicated<F>>>, BoxError> {
+    let converted_shares = try_join_all(zip(repeat(ctx), input).enumerate().map(
+        |(record_id, (ctx, row))| async move {
+            let bit_number = format!("record_id{}", record_id);
+
+            ConvertShares::new(XorShares {
+                num_bits: 40,
```

Comment:
I don't think you should hard-code in this number. This should probably be a param.